matchmaker.js

```diff
@@ -0,0 +1,146 @@
+import Player from './player.js'
+import GameRoom from './gameroom.js';
+import RoomManager from './roommanager.js'
+import { WebSocketServer } from 'ws'
+import { v4 as uuidv4 } from 'uuid';
+
+import { upload_to_blob_storage } from './blobstorage.js';
+
+class Matchmaker {
+    constructor(room_manager) {
+        this.banned_characters = ["carmine"]
+        this.room_manager = room_manager
+        this.match_queues = {}
+
+        this.match_types = {
```

Comment:
Rather than match_types, this is more like queue_definitions

Reply:
I wonder if the queue definitions should be sent to the client on the connect method so the client can populate the UI.
Probably should send down the banned_characters too.
Really, banned characters should probably be inside the queue definition? In case we wanted to have a carmine allowed queue.

Reply:
will do